test/integration/associations/belongs-to-many.test.js

```diff
@@ -557,6 +557,28 @@ describe(Support.getTestDialectTeaser('BelongsToMany'), () => {
       });
     });
 
+    it('compares primary keys using attribute type info', function() {
+      return Promise.all([
+        this.Article.create({ title: 'Article' }),
+        this.Label.create({ text: 'Awesomeness' }),
+        this.Label.create({ text: 'Epicness' })
+      ]).then(([article, label1, label2]) => {
+        return article.setLabels([label1, label2]).then(() => {
+          DataTypes.INTEGER.prototype._sanitize = function _sanitize(value) { return value; };
```

Comment:
If in the future a `_sanitize` function gets added to the `INTEGER` datatype for some reason, you will be overwriting it and deleting it at the end of the test, potentially messing up other tests. Please alter this to make it work safely if a `_sanitize` function exists on the `INTEGER` prototype.

Reply:
fixed